### PR TITLE
Restrict destructive command confirmation to ('yes', 'y')

### PR DESCRIPTION
**Pull request by pasenor**
## Description
Issue #758 
The default boolean converter in Click evaluates "t", "true" and "1" to `True`. Added a custom one.

## Checklist

- [ ] I've added this contribution to the `changelog.md`.
- [ ] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by amjith** [comment]
Should we allow 't', 'true' and '1' to be truthy responses? Somehow I feel like it is not the right approach. I'd much rather have just 'yes' and 'y' as truthy and everything else as negative.

**Comment by pasenor** [comment]
Yeah, but that's exactly what's happening in this PR :)
The default behavior of Click is to allow t and 1, I'm restricting it to only 'y' and 'yes'

File: mycli/packages/prompt_utils.py [hunk 1]
```diff
@@ -7,6 +7,26 @@
 from .parseutils import is_destructive
 
 
+class ConfirmBoolParamType(click.ParamType):
+    name = 'confirmation'
+
+    def convert(self, value, param, ctx):
+        if isinstance(value, bool):
+            return bool(value)
```
Comment (amjith) [inline mycli/packages/prompt_utils.py]: When would we get a boolean value?

**Comment by amjith** [comment]
Looks good to me. :man_juggling:

File: mycli/packages/prompt_utils.py (hunk 1 above)
Comment (pasenor) [reply mycli/packages/prompt_utils.py]: hmm, probably never. I copied it from [here](https://github.com/pallets/click/blob/2c622eed402c64e1645db3904c2410f6c05fbf19/click/types.py#L348) without much thought, maybe we can delete this conditional branch.

**Pull request merged**